library/core/src/array/mod.rs

```diff
@@ -31,6 +31,10 @@ pub use iter::IntoIter;
 /// # Example
 ///
 /// ```rust
+/// // type inference is helping us here, the way `from_fn` knows how many
+/// // elements to produce is the length of array down there: only arrays of
+/// // equal lengths can be compared, so the const generic parameter `N` is
+/// // inferred to be 5, thus creating array of 5 elements.
 /// let array = core::array::from_fn(|i| i);
```

Comment:
Wouldn't this be clearer ?

```suggestion
/// let array: [_; 5] = core::array::from_fn(|i| i);
```

Reply:
Oh right, I wanted to elaborate on this. I think it's better to remind users that Rust has powerful type inference and it is not necessary to provide such type annotations.

Reply:
I would suggest showing this with more examples.

Keep one that infers it, but *also* show different ways to annotating it: `from_fn::<_, N, _>(...)` as well, for example.

(I wish it could be `from_fn::<u8, 10>(…)`, but I lost the fight to use `impl FnMut` here.)